llama-index-core/llama_index/core/schema.py

```diff
@@ -1096,13 +1102,111 @@ def from_cloud_document(
         )
 
 
-class ImageDocument(Document, ImageNode):
-    """Data document containing an image."""
+class ImageDocument(Document):
```

Comment:
We have a few checks in the code like `instance(some_var, ImageNode)` -- I don't THINK this will cause problems, but just wanted to flag

Reply:
uhm this would fail:
```
doc = ImageDocument()
if isinstance(doc, ImageNode):
  do_something
```
I would argue the problem is on the caller, but I'll search for those checks

Reply:
@logan-markewich went through the codebase, this is the only is-instance check that looks problematic https://github.com/run-llama/llama_index/blob/main/llama-index-core/llama_index/core/multi_modal_llms/base.py#L220

The others are related to comparing nodes of different types and won't be affected by this PR

Reply:
I think this check is fine, its a deprecated component anyways (query pipelines)